source/extensions/common/workload_discovery/discovery.proto

```diff
@@ -65,10 +70,10 @@ message Workload {
   reserved "network";
   reserved 4;
 
-  reserved "protocol";
+  reserved "tunnel_protocol";
   reserved 5;
 
-  reserved "waypoint_addresses";
+  reserved "waypoint";
```

Comment:
"protocol" and "waypoint_addresses" are existing fields. Will renaming them impact existing users for these fields? 
An alternative is to add "tunnel_protocol" and "waypoint" as new fields.

Reply:
yes, this is a breaking change.  upgrade is not supported for ztunnel yet

Reply:
No, you cannot do that since we already shipped this code. Never change field numbers or names, even in the experimental protos.

Reply:
Changing names is fine, I thought? Just indexes are the issue, from a proto perspective?

Reply:
We have made it very clear in the development that the API is not stable and has no guarantees - protobuf or otherwise. I don't think there is a need to permanently complicate our APIs because of experimentation we did in the pre-alpha timeframe.

Reply:
This is a different level of compatibility. Data wire compatibility is extremely hard to debug - and if it ever happens that a version of proxy mixes fields/numbers with another istiod, it's going to be impossible to debug. Unfortunately, Istio chose to rely on data-wire compatibility for CRDs which means that the breaking changes in CRDs are breaking on the wire.

@bleggett Yeah, field names are not important, but Envoy will emit a warning when it sees a new name, which bothers some.

Reply:
Experiments can never break production. It's ok to make changes but breaking data plane compatibility (running old Envoy with new istiod) should not be allowed.